lib/puppet/application/lookup.rb

```diff
@@ -379,8 +379,11 @@ def generate_scope
       session = service.create_session
       cert = session.route_to(:ca)
 
-      cert = cert.get_certificate(node)
-      trusted = Puppet::Context::TrustedInformation.new(true, node, cert)
+      _, x509 = cert.get_certificate(node)
+      cert = OpenSSL::X509::Certificate.new(x509)
+
+      Puppet::SSL::Oids.register_puppet_oids
+      trusted = Puppet::Context::TrustedInformation.remote(true, facts.values['certname'] || node, Puppet::SSL::Certificate.from_instance(cert))
 
       Puppet.override(trusted_information: trusted) do
```

Comment:
@joshcooper for this functionality, I'd like to somehow be able to assert that `Puppet.lookup(:trusted_information)` contains the correct custom extensions (parsed from the certificate); do you know of a better way?

Reply:
Looks like you handled this using the `oid.pem` cert, so :+1: from me